### ISSUE TITLE
torch/include/pybind11/eigen.h:12:10: 致命错误：eigen/matrix.h：没有那个文件或目录

### ISSUE THREAD

**Issue by weixiu00** [pytorch/pytorch#112841]
### 🐛 Describe the bug

torch/include/pybind11/eigen/matrix.h   not found

### Versions

torch                    2.1.0+cu121
torchaudio               2.1.0+cu121
torchvision              0.16.0+cu121

cc @seemethere @malfet @osalpekar @atalman

**Comment by bdhirsh** [comment]
It sounds like you're hitting an issue either building or running pytorch with eigen? Do you have a minimal reproduction?

**Comment by malfet** [comment]
Looks like a regression caused by pybind update to 2.11, we just need to package a few more headers, PR is coming.
Marked as regression as with older pybind there were no `eigen` subfolder

**Comment by atalman** [comment]
Lets confirm its regression against 2.0 release

**Comment by atalman** [comment]
confirmed this issue existed in 2.1.0 release we packaged with pybind 2.11.1

**Comment by huydhn** [comment]
Confirm that the issue is fixed in 2.1.2 final RC, so it will be fixed in 2.1.2:

```
$ pip install --pre torch==2.1.2 --index-url https://download.pytorch.org/whl/test
$ ls ~/miniconda3/envs/py3.11/lib/python3.11/site-packages/torch/include/pybind11/eigen
common.h	matrix.h	tensor.h
```
